packages/loopring_v3/contracts/aux/agents/FastWithdrawalAgent2.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2017 Loopring Technology Limited.
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "../../core/iface/IAgentRegistry.sol";
+import "../../core/iface/IExchangeV3.sol";
+import "../../lib/Claimable.sol";
+import "../../lib/EIP712.sol";
+import "../../lib/ERC20SafeTransfer.sol";
+import "../../lib/MathUint.sol";
+import "../../lib/ReentrancyGuard.sol";
+import "../../lib/SignatureUtil.sol";
+import "../../lib/Claimable.sol";
+
+/// @title Fast withdrawal agent implementation. With the help of liquidity providers (LPs),
+///        exchange operators can convert any normal withdrawals into fast withdrawals.
+///
+///        Fast withdrawals are a way for the owner to provide instant withdrawals for
+///        users with the help of a liquidity provider and conditional transfers.
+///
+///        A fast withdrawal requires the non-trustless cooperation of 2 parties:
+///        - A liquidity provider which provides funds to users immediately onchain
+///        - The operator which will make sure the user has sufficient funds offchain
+///          so that the liquidity provider can be paid back.
+///          The operator also needs to process those withdrawals so that the
+///          liquidity provider receives its funds back.
+///
+///        We require the fast withdrawals to be executed by the liquidity provider (as msg.sender)
+///        so that the liquidity provider can impose its own rules on how its funds are spent. This will
+///        inevitably need to be done in close cooperation with the operator, or by the operator
+///        itself using a smart contract where the liquidity provider enforces who, how
+///        and even if their funds can be used to facilitate the fast withdrawals.
+///
+///        The liquidity provider can call `executeFastWithdrawals` to provide users
+///        immediately with funds onchain. This allows the security of the funds to be handled
+///        by any EOA or smart contract.
+///
+///        Users that want to make use of this functionality have to
+///        authorize this contract as their agent.
+///
+/// @author Brecht Devos - <brecht@loopring.org>
+/// @author Kongliang Zhong - <kongliang@loopring.org>
+/// @author Daniel Wang - <daniel@loopring.org>
+contract FastWithdrawalAgent2 is ReentrancyGuard, IAgent, Claimable
+{
+    using AddressUtil       for address;
+    using AddressUtil       for address payable;
+    using ERC20SafeTransfer for address;
+    using MathUint          for uint;
+
+    uint public feeBips = 0;
```

Comment:
Probably better to just make this a parameter in the `Withdrawal` structs. SLOADs already cost 800 gas, but in Berlin (HF in ~6 weeks) will cost [2100 gas](https://eips.ethereum.org/EIPS/eip-2929), so it'll be cheaper to pass onchain each time. Added benefit that the fee can be changed more dynamically (perhaps based on gas costs).